specificatie/zoek-personen.yaml

```diff
@@ -0,0 +1,203 @@
+
+
+        3.  Persoon
+            -  burgerservicenummer
+
+
+        4.  Postcode
+            -  verblijfplaats__postcode
+            -  verblijfplaats__huisnummer
+
+
+        5.  Straat
+            -  verblijfplaats__straat (minimaal 2 karakters, [wildcard toegestaan](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.3.0/features/wildcard.feature) )
+            -  verblijfplaats__gemeenteVanInschrijving
+            -  verblijfplaats__huisnummer
+
+
+        6.  Adres
+            -  verblijfplaats__nummeraanduidingIdentificatie
+
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ZoekPersonenQuery'
+      responses:
+        '200':
+          description: |
+                        Zoekactie geslaagd
+          headers:
+            api-version:
+              $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/headers/api_version"
+            warning:
+              $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/headers/warning"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IngeschrevenPersoonBeperktHalCollectie'
+        '400':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/400"
+        '401':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/401"
+        '403':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/403"
+        '406':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/406"
+        '500':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/500"
+        '501':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/501"
+        '503':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/503"
+        'default':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/default"
+      tags:
+        - Ingeschreven Personen
+components:
+  schemas:
+    ZoekPersonenQuery:
+      type: object
+      required:
+        - fields
```

Comment:
bij het zoeken hoeft fields m.i. niet required te zijn, omdat de response al beperkt is.

Reply:
Volgens mij is deze aanname niet correct. Mag iedereen bijv. geboortedatum opvragen? Volgens mij niet, want dat was de reden om de leeftijd property toe te voegen? Je introduceer volgens mij een privacy lek als een consumer via raadplegen bijv. geen geboortedatum krijgt en via zoeken wel.

Reply:
> Je introduceer volgens mij een privacy lek als een consumer via raadplegen bijv. geen geboortedatum krijgt en via zoeken wel.

Dat zou niet mogen (komt niet succesvol door mijn test heen), want naast de fieldsparameter is er ook nog autorisatie. Je krijgt nooit meer dan waar je voor geautoriseerd bent, niet wanneer je daarom vraagt met fields en ook niet wanneer je geen fields gebruikt. Het maakt dan niet uit of dat via zoeken (de post) of via raadplegen (de get) gaat.

Neemt niet weg dat het misschien toch verstandig kan zijn ook hier fields te verplichten, maar dan om te zorgen dat de gebruiker altijd bewust alleen dat vraagt wat nodig is (werkelijk gebruikt wordt).